cmd/vpu_plugin/vpu_plugin_test.go

```diff
@@ -67,10 +80,27 @@ func TestScan(t *testing.T) {
 		t.Error("vpu plugin test failed")
 	}
 
-	tree, err := testPlugin.scan()
+	var fN fakeNotifier;
+
+	fN.scanDone = testPlugin.scanDone
+	err = testPlugin.Scan(&fN)
 	if err != nil {
 		t.Error("vpu plugin test failed")
 	} else {
-		klog.V(4).Infof("tree len is %d", len(tree[deviceType]))
+		klog.V(4).Infof("tree len is %d", len(fN.tree[deviceType]))
+	}
+
+	//remote the hddl_service.sock and test with no hddl socket case
+	os.Remove("/var/tmp/hddl_service.sock");
```

Comment:
would it make sense to close 'f' here before removing the file?

Reply:
good catch!